transformer_engine/pytorch/dot_product_attention/inference.py

```diff
@@ -128,9 +128,9 @@ def __init__(
         self,
         max_batch_size: int,
         max_sequence_length: int,
-        num_heads_kv: int = 16,
-        head_dim_k: int = 64,
-        dtype: torch.dtype = torch.bfloat16,
+        num_heads_kv: int = None,
+        head_dim_k: int = None,
+        dtype: torch.dtype = None,
```

Comment:
You could also enforce required kwargs with this syntax:
```suggestion
        *,
        num_heads_kv: int,
        head_dim_k: int,
        dtype: torch.dtype,
```